### ISSUE TITLE
TypeScript generics defined on the `script` generic attribute appear to be ignored by the plugin.

### ISSUE THREAD

**Issue by orefalo** [sveltejs/eslint-plugin-svelte#897]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.13.0

### What version of `eslint-plugin-svelte` are you using?

2.46.0

### What did you do?

<details>
TypeScript generics defined on the `script` generic attribute appear to be ignored by the plugin.
</details>

```svelte
// Tree.svelte
<script lang="ts" generics="T">
interface data {
  model: Array<T>;
}
```

### What did you expect to happen?

No errors

### What actually happened?

/.../src/lib/Tree.svelte
   76:18  error  'T' is not defined     no-undef
   77:31  error  'T' is not defined     no-undef
   95:44  error  'T' is not defined     no-undef
   95:60  error  'T' is not defined     no-undef

### Link to **GitHub Repo** with Minimal Reproducible Example

Well, this is a work in progress, but it will do to reproduce the error

1. clone https://github.com/orefalo/svelte-treeviews
2. pnpm i
3. pnpm lint

**Comment by TGlide** [comment]
Any update on this? Any way I can help?

**Comment by TGlide** [comment]
Fix:
```ts
    {
        files: ['**/*.svelte'],
        languageOptions: {
            parser: svelteParser,
            parserOptions: {
                parser: tsEslint.parser,
                svelteFeatures: {
                    experimentalGenerics: true,
                },
            },
        },
    },
    ```

**Comment by orefalo** [comment]
nice, but.. which file is this stored in?
